src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -33,81 +45,314 @@
-        FileUploadSubmission result = fileUploadSubmissionRepository.save(fileUploadSubmission);
-        return ResponseEntity.created(new URI("/api/file-upload-submissions/" + result.getId()))
-                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, result.getId().toString())).body(result);
+    @PostMapping(value = "/exercises/{exerciseId}/file-upload-submissions")
+    @PreAuthorize("hasAnyRole('USER','TA','INSTRUCTOR','ADMIN')")
+    public ResponseEntity<FileUploadSubmission> submitFileUploadExercise(@PathVariable long exerciseId, Principal principal,
+            @RequestPart("submission") FileUploadSubmission fileUploadSubmission, @RequestPart("file") MultipartFile file) {
+        log.debug("REST request to submit new FileUploadSubmission : {}", fileUploadSubmission);
+
+        final var exercise = fileUploadExerciseService.findOne(exerciseId);
+        if (!authCheckService.isAtLeastStudentForExercise(exercise)) {
+            return forbidden();
+        }
+
+        // Check if the course hasn't been changed
+        final var validityExceptionResponse = this.checkExerciseValidity(exercise);
+        if (validityExceptionResponse != null) {
+            return validityExceptionResponse;
+        }
+
+        // Check the file size
+        if (file.getSize() > Constants.MAX_UPLOAD_FILESIZE_BYTES) {
+            // NOTE: Maximum file size for submission is 2 MB
+            return ResponseEntity.status(413).headers(HeaderUtil.createAlert(applicationName, "The maximum file size is 2MB!", "fileUploadSubmissionFileTooBig")).build();
+        }
+
+        // Check the pattern
+        final var splittedFileName = file.getOriginalFilename().split("\\.");
+        final var fileSuffix = splittedFileName[splittedFileName.length - 1].toLowerCase();
+        final var filePattern = String.join("|", exercise.getFilePattern().toLowerCase().replace(" ", "").split(","));
+        if (!fileSuffix.matches(filePattern)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .headers(HeaderUtil.createAlert(applicationName, "The uploaded file has the wrong type!", "fileUploadSubmissionIllegalFileType")).build();
+        }
+
+        final FileUploadSubmission submission;
+        try {
+            submission = fileUploadSubmissionService.handleFileUploadSubmission(fileUploadSubmission, file, exercise, principal);
+        }
+        catch (IOException e) {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(applicationName, true, "fileUploadSubmission", "fileUploadSubmissionCantStore",
+                    "The uploaded file could not be saved on the server")).build();
+        }
+
+        hideDetails(submission);
+        return ResponseEntity.ok(submission);
     }
 
     /**
-     * PUT /file-upload-submissions : Updates an existing fileUploadSubmission.
+     * GET /file-upload-submissions/:id : get the "id" fileUploadSubmission.
```

Comment:
A documentation is missing for what we need this API call.
Is it for tutors?

Reply:
Yes, it is for tutors assessment, changed the description